File: src/main/java/no/nav/permitteringsskjemaapi/kodeverk/KodeverkController.java
package no.nav.permitteringsskjemaapi.kodeverk;

import no.nav.permitteringsskjemaapi.permittering.PermitteringsskjemaType;
import no.nav.permitteringsskjemaapi.permittering.Årsakskode;
import no.nav.security.token.support.core.api.Unprotected;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.EnumMap;
import java.util.Map;

@Unprotected
@RestController
@RequestMapping("/kodeverk")
public class KodeverkController {
    @GetMapping
    public Map<String, Map<? extends Enum, String>> alleKoder() {
        return Map.of(
                "årsakskoder", årsakskoder(),
                "skjematyper", skjematyper()
        );
    }

    @GetMapping("/årsakskoder")
    public Map<Årsakskode, String> årsakskoder() {
        var map = new EnumMap<Årsakskode, String>(Årsakskode.class);
        for (Årsakskode årsakskode : Årsakskode.values()) {
            map.put(årsakskode, årsakskode.getNavn());
        }
        return map;
    }

    @GetMapping("/skjematyper")
    public Map<PermitteringsskjemaType, String> skjematyper() {
        var map = new EnumMap<PermitteringsskjemaType, String>(PermitteringsskjemaType.class);
        for (var skjemaType : PermitteringsskjemaType.values()) {
            map.put(skjemaType, skjemaType.getNavn());
        }
        return map;
    }
}

File: src/main/kotlin/no/nav/permitteringsskjemaapi/kodeverk/KodeverkController.kt
package no.nav.permitteringsskjemaapi.kodeverk

import io.swagger.v3.oas.annotations.Hidden
import no.nav.permitteringsskjemaapi.permittering.PermitteringsskjemaType
import no.nav.permitteringsskjemaapi.permittering.Årsakskode
import no.nav.security.token.support.core.api.Unprotected
import org.springframework.web.bind.annotation.*
import java.util.*

@Unprotected
@RestController
@RequestMapping("/kodeverk")
class KodeverkController {
    @Hidden
    @GetMapping
    fun alleKoder() = mapOf(
        "årsakskoder" to årsakskoder(),
        "skjematyper" to skjematyper()
    )

    @GetMapping("/årsakskoder")
    fun årsakskoder(): Map<Årsakskode?, String?> {
        val map = EnumMap<Årsakskode, String?>(Årsakskode::class.java)
        for (årsakskode in Årsakskode.values()) {
            map[årsakskode] = årsakskode.navn
        }
        return map
    }

    @GetMapping("/skjematyper")
    fun skjematyper(): Map<PermitteringsskjemaType?, String?> {
        val map = EnumMap<PermitteringsskjemaType, String>(
            PermitteringsskjemaType::class.java
        )
        for (skjemaType in PermitteringsskjemaType.values()) {
            map[skjemaType] = skjemaType.navn
        }
        return map
    }
}